tapchannel/aux_invoice_manager_test.go

```diff
@@ -0,0 +1,571 @@
+	request.ExitHtlcAmt = lnwire.MilliSatoshi(request.Invoice.ValueMsat)
+
+	return request, assetUnits, rfqID
+}
+
+// genRequests generates a random array of requests to be processed by the
+// AuxInvoiceManager. It also returns the rfq map with the related rfq quotes.
+func genRequests(t *rapid.T) ([]lndclient.InvoiceHtlcModifyRequest,
+	rfq.BuyAcceptMap) {
+
+	rfqMap := rfq.BuyAcceptMap{}
+
+	len := rapid.IntRange(0, 5).Draw(t, "requestsLen")
+	requests := make([]lndclient.InvoiceHtlcModifyRequest, len)
+
+	for i := range requests {
+		req, numAssets, scid := genRequest(t)
+		requests[i] = req
+
+		genBuyQuotes(
+			t, rfqMap, numAssets, uint64(req.Invoice.ValueMsat),
+			scid,
+		)
+	}
+
+	return requests, rfqMap
+}
+
+// genBuyQuotes populates the provided map of rfq quotes with the desired values
+// for a specific
+func genBuyQuotes(t *rapid.T, rfqMap rfq.BuyAcceptMap, units, amtMsat uint64,
+	scid rfqmsg.ID) {
+
+	var peer route.Vertex
+	askPrice := lnwire.MilliSatoshi(0)
+
+	// Introduce a 1/8 chance that the quote's peerID is not correct.
+	noPeerMatch := rapid.Uint16().Draw(t, "nodeID_mismatch_probability")
+	if noPeerMatch%8 == 0 {
+		peer = route.Vertex{3, 1, 4}
+	} else {
+		peer = testNodeID
+	}
+
+	// Introduce a 1/5 chance that the askPrice of this asset will result in
+	// a random total asset value.
+	noValueMatch := rapid.Uint16().Draw(t, "no_asset_value_match")
+	if noValueMatch%5 == 0 {
+		askPrice = lnwire.MilliSatoshi(
+			rapid.Uint64Range(
+				0, 250_000,
+			).Draw(t, "asset_msat_value"))
+	} else {
+		askPrice = lnwire.MilliSatoshi(amtMsat / units)
+	}
+
+	rfqMap[scid.Scid()] = rfqmsg.BuyAccept{
+		Peer:     peer,
+		AskPrice: askPrice,
+	}
+}
+
+// testInvoiceManager creates an array of requests to be processed by the
+// AuxInvoiceManager. Uses the enhanced HtlcMmodifierMockProperty instance.
+func testInvoiceManager(t *rapid.T) {
+	requests, rfqMap := genRequests(t)
+
+	mockRfq := &mockRfqManager{
+		peerBuyQuotes: rfqMap,
+	}
+
+	done := make(chan bool)
+
+	mockModifier := &mockHtlcModifierProperty{
+		requestQue: requests,
+		rfqMap:     rfqMap,
+		done:       done,
+		t:          t,
+	}
+
+	manager := NewAuxInvoiceManager(
+		&InvoiceManagerConfig{
+			ChainParams:         testChainParams,
+			InvoiceHtlcModifier: mockModifier,
+			RfqManager:          mockRfq,
+		},
+	)
+
+	err := manager.Start()
+	require.NoError(t, err)
+
+	select {
+	case <-done:
+	case <-time.After(testTimeout):
+		t.Fail()
+	}
+}
+
+// TestAuxInvoiceManagerProperty runs property based tests on the
+// AuxInvoiceManager.
+func TestAuxInvoiceManagerProperty(t *testing.T) {
+	t.Parallel()
+
+	t.Run("invoice_manager", rapid.MakeCheck(testInvoiceManager))
+}
+
+func newHash(i []byte) []byte {
+	h := sha256.New()
+	_, _ = h.Write(i)
+
+	return h.Sum(nil)
+}
+
+func assetID(i byte) asset.ID {
+	return asset.ID(newHash([]byte{i}))
+}
+
+func dummyRfqID(value int) rfqmsg.ID {
+	var rfqID rfqmsg.ID
+	for i := 0; i < len(rfqID); i++ {
```

Comment:
super-nit: could use `bytes.Repeat()` here.

Reply:
true, casting between types makes this into more diff though